template.yaml

```diff
@@ -599,3 +602,91 @@ Resources:
       DomainName: !Ref ApiDomain
       RestApiId: !Ref NvaCristinProxyApi
       Stage: !Ref NvaCristinProxyApi.Stage
+
+  NvaCristinIntermediateStorageBucket:
+    Type: AWS::S3::Bucket
+    DeletionPolicy: Delete
+    Properties:
+      LifecycleConfiguration:
+        Rules:
+          - Status: 'Enabled'
+            ExpirationInDays: 7
+            Id: 'DeleteFilesAfter7Days'
+
+  NvaCristinIntermediateStorageBucketPolicy:
+    Type: 'AWS::S3::BucketPolicy'
+    Properties:
+      Bucket: !Ref NvaCristinIntermediateStorageBucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              AWS: !Sub '${AWS::AccountId}'
+            Action:
+              - "s3:GetObject"
+            Resource: !Sub 'arn:aws:s3:::${NvaCristinIntermediateStorageBucket}/units-norway.json'
+
+  NvaCristinIntermediateStorageUnitsParameter:
+    Type: 'AWS::SSM::Parameter'
+    DependsOn: NvaCristinIntermediateStorageBucket
+    Condition: IsCustomDomainBasePathCristin
+    Properties:
+      Description: 'S3 URI of the file in S3 bucket'
+      Name: '/NVA/S3/CristinIntermediateStorage/Units'
+      Type: 'String'
+      Value: !Sub 's3://${NvaCristinIntermediateStorageBucket}/units-norway.json'
+
+  NvaCristinIntermediateStorageBucketWritePolicy:
+    Type: 'AWS::IAM::ManagedPolicy'
+    DependsOn: NvaCristinIntermediateStorageBucket
+    Properties:
+      Description: 'Managed policy for writing objects to S3 from Lambda'
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Action:
+              - "s3:PutObject"
+            Resource: !Sub 'arn:aws:s3:::${NvaCristinIntermediateStorageBucket}/*'
+
+  NvaCristinUpdateIntermediateUnitsFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: cristin-intermediate-storage
+      Handler: no.unit.nva.cristin.intermediate.storage.UpdateUnitsHandler::handleRequest
+      Policies:
+        - !Ref NvaCristinIntermediateStorageBucketWritePolicy
+      Environment:
+        Variables:
+          INTERMEDIATE_STORAGE_BUCKET_NAME: !Ref NvaCristinIntermediateStorageBucket
+          CRISTIN_BOT_FILTER_BYPASS_HEADER_NAME: !Ref CristinBotFilterBypassHeaderName
+          CRISTIN_BOT_FILTER_BYPASS_HEADER_VALUE: !Ref CristinBotFilterBypassHeaderValue
+          CRISTIN_API_URL: !Ref CristinApiUrl
+      Events:
+        Schedule:
+          Type: Schedule
+          Properties:
+            Schedule: 'cron(0 5,12 * * ? *)'
+
+  NvaCristinUpdateIntermediateUnitsFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub '/aws/lambda/${NvaCristinUpdateIntermediateUnitsFunction}'
+      RetentionInDays: 90
```

Comment:
Will this be deleted after each karate test run, or will they stack?

Reply:
looks like it actually deletes it, but if not contents will be deleted in 90 days at least